stress_benchmark/stress_benchmark.cpp

```diff
@@ -0,0 +1,257 @@
+    }
+    cura::LayerIndex layer_idx(100);
+    cura::WallsComputation walls_computation(settings, layer_idx);
+    walls_computation.generateWalls(&layer, cura::SectionType::WALL);
+    exit(EXIT_SUCCESS);
+}
+
+size_t checkCrashCount(size_t crashCount, int status, const auto& resource)
+{
+    if (WIFSIGNALED(status))
+    {
+        ++crashCount;
+        spdlog::error("Crash detected for: {}", resource.stem());
+    }
+    return crashCount;
+}
+
+rapidjson::Value
+    createRapidJSONObject(rapidjson::Document::AllocatorType& allocator, const std::string& test_name, const auto value, const std::string& unit, const std::string& extra_info)
+{
+    rapidjson::Value obj(rapidjson::kObjectType);
+    rapidjson::Value key("name", allocator);
+    rapidjson::Value val1(test_name.c_str(), test_name.length(), allocator);
+    obj.AddMember(key, val1, allocator);
+    key.SetString("unit", allocator);
+    rapidjson::Value val2(unit.c_str(), unit.length(), allocator);
+    obj.AddMember(key, val2, allocator);
+    key.SetString("value", allocator);
+    rapidjson::Value val3(value);
+    obj.AddMember(key, val3, allocator);
+    key.SetString("extra", allocator);
+    rapidjson::Value val4(extra_info.c_str(), extra_info.length(), allocator);
+    obj.AddMember(key, val4, allocator);
+    return obj;
+}
+
+void createAndWriteJson(const std::filesystem::path& out_file, double stress_level, const std::string& extra_info, const size_t no_test_cases)
+{
+    rapidjson::Document doc;
+    doc.SetArray();
+    rapidjson::Document::AllocatorType& allocator = doc.GetAllocator();
+    auto no_test_cases_obj = createRapidJSONObject(allocator, "Number of test cases", no_test_cases, "-", "");
+    doc.PushBack(no_test_cases_obj, allocator);
+
+    auto stress_obj = createRapidJSONObject(allocator, "General Stress Level", stress_level, "%", extra_info);
+    doc.PushBack(stress_obj, allocator);
+
+    rapidjson::StringBuffer buffer;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    doc.Accept(writer);
+
+    spdlog::info("Writing Json results: {}", std::filesystem::absolute(out_file).string());
+    std::ofstream file{ out_file };
+    if (! file)
+    {
+        spdlog::critical("Failed to open the file: {}", out_file.string());
+        exit(EXIT_FAILURE);
+    }
+    file.write(buffer.GetString(), buffer.GetSize());
+    file.close();
+}
+
+int main(int argc, const char** argv)
+{
+    constexpr bool show_help = true;
+    constexpr std::string_view version = "0.1.0";
+    const std::map<std::string, docopt::value> args = docopt::docopt(fmt::format("{}", USAGE), { argv + 1, argv + argc }, show_help, fmt::format("{}", version));
+
+    const auto resources = getResources();
+    size_t crash_count = 0;
+    std::vector<std::string> extra_infos;
+
+    for (const auto& resource : resources)
+    {
+        const auto& shapes = resource.polygons();
+        const auto& settings = resource.settings();
+
+        pid_t pid = fork();
```

Comment:
yeah,  I know I tried to do this differently first, more OS agnostic but that introduced a lot of extra complexity and edge cases to take care of.

It might be best if we remove the stress benchmark option from the conanfile in the engine for Windows MSVC

Reply:
Yes, good idea, that way we also indicate that this isn't supported by us on that.